src/scenes/MainScene.ts

```diff
@@ -0,0 +1,48 @@
+import Phaser from 'phaser';
+import { generateHills } from '@src/map/generateHills';
+import Ball from '@src/entities/Ball';
+import Goal from '@src/entities/Goal';
+import Ground from '@src/entities/Ground';
+
+export default class MainScene extends Phaser.Scene {
+  goal?: Goal;
+  ball?: Ball;
+  ground?: Ground;
+
+  constructor() {
+    super({ key: 'MainScene' });
+  }
+
+  preload() {
+    this.load.image('goal', 'assets/goalflag.png');
+  }
+
+  create() {
+    const hills = generateHills(
+      this.game.config.width as number,
+      this.game.config.height as number
+    );
+
+    this.ground = new Ground(this, hills);
+    this.ball = new Ball(this);
+    this.goal = new Goal(this);
+
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    this.ball.setOnCollideWith(this.goal, () => {
+      this.endGame();
+    });
+
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    this.goal.setOnCollideWith(this.matter.world.getAllBodies(), () => {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      this.goal.body.isStatic = true;
+    });
+  }
```

Comment:
digging through some old stackoverflows and github issues I found those but TS doesn't like them

.body.isStatic - https://github.com/photonstorm/phaser3-docs/issues/24
.setOnCollideWith - https://stackoverflow.com/questions/50223192/phaser-3-matter-physics-detect-collision

the more I toy around with phaser the less I like it, things feel weirdly complicated for no apparent reason

Reply:
Since this comment I looked further into the types. It does kind of work
But I didn't find any documentation

Reply:
I was using the word.on("collisionstart") previously but I really disliked how the code looks like